File: app/src/main/java/hu/pe/biko/biko/RoutesAdapter.java
package hu.pe.biko.biko;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

/**
 * Created by nikita on 27.05.17.
 */

public class RoutesAdapter extends RecyclerView.Adapter<RoutesAdapter.MyViewHolder> {

    private List<Route> routeList;

    public RoutesAdapter(List<Route> routes) { routeList = routes; }

    @Override
    public RoutesAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.route_card, parent, false);
        return new RoutesAdapter.MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(final RoutesAdapter.MyViewHolder holder, int position) {
        Route route = routeList.get(position);
    }

    @Override
    public int getItemCount() { return routeList.size(); }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        public View view;
        public ImageView image;
        public TextView name;
        public TextView description;

        public  MyViewHolder(View itemView) {
            super(itemView);
            view = itemView;
            name = (TextView) itemView.findViewById(R.id.name_card_routes);
            description = (TextView) itemView.findViewById(R.id.shortdescription_card_routes);
            image = (ImageView) itemView.findViewById(R.id.img_card_main);
        }

    }
}

File: app/src/main/java/hu/pe/biko/biko/RoutesAdapter.kt
package hu.pe.biko.biko

import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide

class RoutesAdapter(private val routeList: List<Route>) : RecyclerView.Adapter<RoutesAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RoutesAdapter.MyViewHolder {

        val v = LayoutInflater.from(parent.context).inflate(R.layout.route_card, parent, false)
        return RoutesAdapter.MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: RoutesAdapter.MyViewHolder, position: Int) {
        val route = routeList[position]
        holder.name.text = route.name
        holder.description.text = route.description
        Glide.with(holder.view.context)
                .load(route.image)
                .error(R.drawable.caption)
                .placeholder(R.drawable.caption)
                .into(holder.image)
        holder.view.setOnClickListener { v ->
            val intent = Intent(holder.view.context, RouteActivity::class.java)
            intent.putExtra("route", route)
            holder.view.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return routeList.size
    }

    class MyViewHolder(var view: View) : RecyclerView.ViewHolder(view) {
        var image: ImageView
        var name: TextView
        var description: TextView

        init {
            name = view.findViewById(R.id.name_card_routes) as TextView
            description = view.findViewById(R.id.shortdescription_card_routes) as TextView
            image = view.findViewById(R.id.img_card_main) as ImageView
        }
    }

}
